apps/studio/components/layouts/SQLEditorLayout/SQLEditorNavV2/SQLEditorNav.tsx

```diff
@@ -42,12 +44,19 @@ interface SQLEditorNavProps {
   sort?: 'inserted_at' | 'name'
 }
 
+type SectionState = { shared: boolean; favorite: boolean; private: boolean }
+const DEFAULT_SECTION_STATE: SectionState = { shared: false, favorite: false, private: false }
+
 export const SQLEditorNav = ({ sort = 'inserted_at' }: SQLEditorNavProps) => {
   const router = useRouter()
   const { profile } = useProfile()
   const project = useSelectedProject()
   const { ref: projectRef, id } = useParams()
   const snapV2 = useSqlEditorV2StateSnapshot()
+  const [sections, setSections] = useLocalStorage<SectionState | undefined>(
```

Comment:
We should avoid duplicating state and remove
```
const [showFavoriteSnippets, setShowFavoriteSnippets] = useState(false)
const [showSharedSnippets, setShowSharedSnippets] = useState(false)
const [showPrivateSnippets, setShowPrivateSnippets] = useState(true)
```
we can write instead
```
const [{shared: showSharedSnippets, favorite: showFavoriteSnippets, private: showPrivateSnippets }, setSections] = useLocalStorage
```

Reply:
very good point! 🙏